sigauth/signature.go

```diff
@@ -116,11 +139,15 @@ func (s *SignatureAuth) SignRequest(r *http.Request) error {
 
 	headers := []string{"request-line", "host", "date", "digest", "content-length"}
 
-	serviceKeyFingerprint, err := authutils.GetKeyFingerprint(&s.serviceKey.PublicKey)
+	publicKey, ok := s.serviceKey.Public().(authutils.PublicKey)
+	if !ok {
+		return errors.New("unrecognized service key type")
+	}
+	keyID, err := authutils.GetKeyFingerprint(publicKey)
 	if err != nil {
 		return fmt.Errorf("error getting service key fingerprint: %v", err)
 	}
-	sigAuthHeader := SignatureAuthHeader{KeyID: serviceKeyFingerprint, Algorithm: "rsa-sha256", Headers: headers}
+	sigAuthHeader := SignatureAuthHeader{KeyID: keyID, Algorithm: "rsa-sha256", Headers: headers}
```

Comment:
The `Algorithm` field needs to be set correctly for the provided key type used to create the signature.